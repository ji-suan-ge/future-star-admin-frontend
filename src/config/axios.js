import Vue from 'vue'
import axios from 'axios'
import VueAxios from 'vue-axios'
import globalConstant from '../constant/globalConstant'

Vue.use(VueAxios, axios)

axios.defaults.baseURL = globalConstant.host
axios.defaults.withCredentials = true
