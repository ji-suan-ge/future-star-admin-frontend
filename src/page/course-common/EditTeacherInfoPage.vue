<template>
  <el-container>
    <el-main>
      <el-page-header @back="goBack()" content="讲师信息"></el-page-header>
      <el-divider/>
      <el-card shadow="never">
        <el-form>
          <el-row>
            <el-form-item align="center">
              <el-upload
                class="avatar-uploader"
                action="https://jsonplaceholder.typicode.com/posts/"
                :show-file-list="false"
                :http-request="uploadImg"
                :on-success="uploadImgSuccess">
                <img v-if="teacher.avatar" :src="teacher.avatar"  width=240px class="image" alt="image">
                <i v-else class="el-icon-plus avatar-uploader-icon">添加头像</i>
              </el-upload>
            </el-form-item>
            <el-form-item label="讲师名：">
              <el-input v-model="teacher.name"></el-input>
            </el-form-item>
          </el-row>
          <el-row>
            <el-form-item label="讲师头衔：">
              <el-input v-model="teacher.title"></el-input>
            </el-form-item>
          </el-row>
          <el-row>
            <el-form-item label="讲师联系方式：">
              <el-input v-model="teacher.contact_way"></el-input>
            </el-form-item>
          </el-row>
          <el-row>
            <el-form-item label="讲师简介：">
              <el-input type="textarea" v-model="teacher.introduction"
                        :autosize="{ minRows: 2, maxRows: 8}"></el-input>
            </el-form-item>
          </el-row>
          <el-form-item>
            <el-row type="flex" justify="space-around">
              <el-col :span="3">
                <el-button type="primary" @click="saveTeacherInfo" round>保存</el-button>
              </el-col>
              <el-col :span="3">
                <el-button @click="clearText" round>清空</el-button>
              </el-col>
            </el-row>
          </el-form-item>
        </el-form>
      </el-card>
    </el-main>
  </el-container>
</template>

<script>
export default {
  name: 'EditTeacherInfoPage',
  data () {
    return {
      teacher: {}
    }
  },
  created () {
    if (this.$store.getters.getCurrentCourse) {
      let course = this.$store.getters.getCurrentCourse
      this.teacher = course.teacher
    }
  },
  methods: {
    goBack: function () {
      this.$router.go(-1)
    },
    uploadImg (f) {
      let param = new FormData()
      let url = '/file/upload'
      param.append('file', f.file)
      let config = {
        headers: { 'Content-Type': 'multipart/form-data' }
      }
      this.axios.post(url, param, config)
        .then(response => {
          f.onSuccess(response.data)
        })
        .catch(() => {
          f.onError()
        })
    },
    uploadImgSuccess (response, file, fileList) {
      this.teacher.avatar = response.data.url
      this.$message({
        showClose: true,
        message: '修改讲师头像成功',
        type: 'success' })
    },
    clearText: function () {
      this.teacher.name = ''
      this.teacher.avatar = ''
      this.teacher.title = ''
      this.teacher.introduction = ''
      this.teacher.contact_way = ''
    },
    saveTeacherInfo: function () {
      let that = this
      let url = '/course/teacher/' + that.teacher.id
      that.axios.put(url, {
        name: that.teacher.name,
        avatar: that.teacher.avatar,
        title: that.teacher.title,
        introduction: that.teacher.introduction,
        contact_way: that.teacher.contact_way
      }).then(function (response) {
        if (response.data.code === '2000') {
          that.$message({
            type: 'success',
            message: '保存成功'
          })
        } else {
          that.$message({
            type: 'error',
            message: '服务器繁忙，请稍后重试'
          })
        }
      }).catch(function (error) {
        console.log(error)
        that.$message({
          type: 'error',
          message: '服务器繁忙，请稍后重试'
        })
      })
    }
  }
}
</script>

<style scoped>
</style>
