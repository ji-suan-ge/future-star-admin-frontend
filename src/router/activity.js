import CreateActivityPage from '../page/CreateActivityPage'
import ShowActivityParticipantsPage from '../page/ShowActivityParticipantsPage'
import EditExistingActivityPage from '../page/EditExistingActivityPage'
import ShowActivityInformationPage from '../page/ShowActivityInformationPage'

export default [
  {
    path: '/create-activity',
    name: 'CreateActivityPage',
    component: CreateActivityPage
  },
  {
    path: '/show-participants',
    name: 'ShowActivityParticipantsPage',
    component: ShowActivityParticipantsPage
  },
  {
    path: '/edit-activity',
    name: 'EditExistingActivityPage',
    component: EditExistingActivityPage
  },
  {
    path: '/show-activity-details',
    name: 'ShowActivityInformationPage',
    component: ShowActivityInformationPage
  }
]
