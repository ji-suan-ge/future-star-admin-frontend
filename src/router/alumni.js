import ShowAlumniInformationPage from '../page/alumni-management/ShowAlumniInformationPage'
import EditAlumniInformationPage from '../page/alumni-management/EditAlumniInformationPage'
import ShowAlumniEditHistoryPage from '../page/alumni-management/ShowAlumniEditHistoryPage'
import ShowAlumniHistoryInformationPage from '../page/alumni-management/ShowAlumniHistoryInformationPage'

export default [
  {
    path: '/show-all-information',
    name: 'ShowAlumniInformationPage',
    component: ShowAlumniInformationPage
  },
  {
    path: '/edit-alumni-information',
    name: 'EditAlumniInformationPage',
    component: EditAlumniInformationPage
  },
  {
    path: '/show-edit-history',
    name: 'ShowAlumniEditHistoryPage',
    component: ShowAlumniEditHistoryPage
  },
  {
    path: '/show-history-information',
    name: 'ShowAlumniHistoryInformationPage',
    component: ShowAlumniHistoryInformationPage
  }
]
