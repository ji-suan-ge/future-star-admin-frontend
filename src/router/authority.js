import AddAdministratorPage from '../page/authority-management/AddAdministratorPage'

export default [
  {
    path: '/add-administrator',
    name: 'AddAdministratorPage',
    component: AddAdministratorPage
  }
]
