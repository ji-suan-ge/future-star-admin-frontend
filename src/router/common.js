import LoginPage from '../page/LoginPage'
import HomePage from '../page/HomePage'
import homeChildren from './home-children'

export default [
  {
    path: '/',
    name: 'Root',
    redirect: '/login'
  },
  {
    path: '/login',
    name: 'LoginPage',
    component: LoginPage
  },
  {
    path: '/home',
    name: 'HomePage',
    component: HomePage,
    redirect: '/home/welcome',
    children: [
      ...homeChildren
    ]
  }
]
