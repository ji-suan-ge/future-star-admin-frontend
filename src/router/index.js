import Vue from 'vue'
import VueRouter from 'vue-router'
import LoginPage from '../page/LoginPage'
import HomePage from '../page/HomePage'
import EnrollmentManagementPage from '../page/EnrollmentManagementPage'
import CourseManagementPage from '../page/CourseManagementPage'
import ActivityManagementPage from '../page/ActivityManagementPage'
import AlumniManagementPage from '../page/AlumniManagementPage'
import AuthorityManagementPage from '../page/AuthorityManagementPage'
import CreateActivityPage from '../page/CreateActivityPage'
import AddAdministrator from '../page/AddAdministrator'
import WelcomePage from '../page/WelcomPage'

Vue.use(VueRouter)

const routes = [
  {
    path: '/',
    name: 'Root',
    redirect: '/login'
  },
  {
    path: '/login',
    name: 'LoginPage',
    component: LoginPage
  },
  {
    path: '/home',
    name: 'HomePage',
    component: HomePage,
    redirect: '/home/welcome',
    children: [
      {
        path: 'welcome',
        name: 'WelcomePage',
        component: WelcomePage
      },
      {
        path: 'enrollment-management',
        name: 'EnrollmentManagementPage',
        component: EnrollmentManagementPage
      },
      {
        path: 'course-management',
        name: 'CourseManagementPage',
        component: CourseManagementPage
      },
      {
        path: 'activity-management',
        name: 'ActivityManagementPage',
        component: ActivityManagementPage
      },
      {
        path: 'alumni-management',
        name: 'AlumniManagementPage',
        component: AlumniManagementPage
      },
      {
        path: 'authority-management',
        name: 'AuthorityManagementPage',
        component: AuthorityManagementPage
      },
      {
        path: 'add-administrator',
        name: 'AddAdministrator',
        component: AddAdministrator
      }
    ]
  },
  {
    path: '/create-activity',
    name: 'CreateActivityPage',
    component: CreateActivityPage
  }
]

const router = new VueRouter({
  routes
})

export default router
