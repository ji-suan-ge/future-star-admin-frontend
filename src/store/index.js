import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    activeIndexOfNavigation: '0',
    activeNameOfCourse: 'first',
    activeNameOfEnrollment: 'first',
    activeNameOfActivity: 'first',
    activeSemesterOfCourse: '',
    semesters: '',
    currentSemester: '',
    classes: '',
    currentClass: '',
    courses: '',
    currentCourse: '',
    infoOfAnAlumni: null
  },
  mutations: {
    changeActiveIndexOfNavigation (state, activeIndex) {
      state.activeIndexOfNavigation = activeIndex
    },
    changeActiveNameOfEnrollment (state, activeName) {
      state.activeNameOfEnrollment = activeName
    },
    changeActiveNameOfCourse (state, activeName) {
      state.activeNameOfCourse = activeName
    },
    changeActiveNameOfActivity (state, activeName) {
      state.activeNameOfActivity = activeName
    },
    changeActiveSemesterOfCourse (state, activeID) {
      state.activeSemesterOfCourse = activeID
    },
    changeSemesters (state, semesters) {
      state.semesters = semesters
    },
    changeInfoOfAnAlumni (state, alumniInfo) {
      state.infoOfAnAlumni = alumniInfo
    },
    changeCurrentSemester (state, semester) {
      state.currentSemester = semester
    },
    changeClasses (state, classes) {
      state.classes = classes
    },
    changeCurrentClass (state, currentClass) {
      state.currentClass = currentClass
    },
    changeCourses (state, courses) {
      state.courses = courses
    },
    changeCurrentCourse (state, currentCourse) {
      state.currentCourse = currentCourse
    },
    initializeStateWhileLogin (state) {
      state.activeIndexOfNavigation = '0'
      state.activeNameOfCourse = 'first'
      state.activeNameOfEnrollment = 'first'
      state.activeNameOfActivity = 'first'
      state.activeSemesterOfCourse = ''
      state.semesters = ''
      state.currentSemester = ''
      state.classes = ''
      state.currentClass = ''
      state.courses = ''
      state.currentCourse = ''
      state.infoOfAnAlumni = null
    },
    initializeStateWhileSwitch (state) {
      state.activeNameOfCourse = 'first'
      state.activeNameOfEnrollment = 'first'
      state.activeNameOfActivity = 'first'
      state.activeSemesterOfCourse = ''
      state.semesters = ''
      state.currentSemester = ''
      state.classes = ''
      state.currentClass = ''
      state.courses = ''
      state.currentCourse = ''
      state.infoOfAnAlumni = null
    }
  },
  actions: {
    changeActiveIndexOfNavigation (context, activeIndex) {
      context.commit('changeActiveIndexOfNavigation', activeIndex)
    },
    changeActiveNameOfEnrollment (context, activeName) {
      context.commit('changeActiveNameOfEnrollment', activeName)
    },
    changeActiveNameOfCourse (context, activeName) {
      context.commit('changeActiveNameOfCourse', activeName)
    },
    changeActiveNameOfActivity (context, activeName) {
      context.commit('changeActiveNameOfActivity', activeName)
    },
    changeActiveSemesterOfCourse (context, activeID) {
      context.commit('changeActiveSemesterOfCourse', activeID)
    },
    changeSemesters (context, semesters) {
      context.commit('changeSemesters', semesters)
    },
    changeInfoOfAnAlumni (context, alumniInfo) {
      context.commit('changeInfoOfAnAlumni', alumniInfo)
    },
    changeCurrentSemester (context, semester) {
      context.commit('changeCurrentSemester', semester)
    },
    changeClasses (context, classes) {
      context.commit('changeClasses', classes)
    },
    changeCurrentClass (context, currentClass) {
      context.commit('changeCurrentClass', currentClass)
    },
    changeCourses (context, courses) {
      context.commit('changeCourses', courses)
    },
    changeCurrentCourse (context, currentCourse) {
      context.commit('changeCurrentCourse', currentCourse)
    },
    initializeStateWhileLogin (context) {
      context.commit('initializeStateWhileLogin')
    },
    initializeStateWhileSwitch (context) {
      context.commit('initializeStateWhileSwitch')
    }
  },
  getters: {
    getActiveIndexOfNavigation (state) {
      return state.activeIndexOfNavigation
    },
    getActiveNameOfEnrollment (state) {
      return state.activeNameOfEnrollment
    },
    getActiveNameOfCourse (state) {
      return state.activeNameOfCourse
    },
    getActiveNameOfActivity (state) {
      return state.activeNameOfActivity
    },
    getActiveSemesterOfCourse (state) {
      return state.activeSemesterOfCourse
    },
    getSemesters (state) {
      return state.semesters
    },
    getInfoOfAnAlumni (state) {
      return state.infoOfAnAlumni
    },
    getCurrentSemester (state) {
      return state.currentSemester
    },
    getClasses (state) {
      return state.classes
    },
    getCurrentClass (state) {
      return state.currentClass
    },
    getCourses (state) {
      return state.courses
    },
    getCurrentCourse (state) {
      return state.currentCourse
    }
  },
  modules: {
  }
})
