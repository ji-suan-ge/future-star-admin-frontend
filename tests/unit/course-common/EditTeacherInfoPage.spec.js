import { createLocalVue, mount } from '@vue/test-utils'
import EditTeacherInfoPage from '../../../src/page/course-common/EditTeacherInfoPage'
import Vuex from 'vuex'
import ElementUI from 'element-ui'
import VueRouter from 'vue-router'
import VueAxios from 'vue-axios'
import axios from 'axios'
import store from '../../../src/store'
import router from '../../../src/router'

describe('EditTeacherInfoPage', () => {
  const localVue = createLocalVue()
  localVue.use(Vuex)
  localVue.use(ElementUI)
  localVue.use(VueRouter)
  localVue.use(VueAxios, axios)
  const wrapper = mount(EditTeacherInfoPage, {
    store,
    router,
    localVue
  })
  test('快照测试', async () => {
    await expect(wrapper.html()).toMatchSnapshot()
  })
  test('是一个 Vue 实例', async () => {
    await expect(wrapper.isVueInstance()).toBeTruthy()
  })
})
