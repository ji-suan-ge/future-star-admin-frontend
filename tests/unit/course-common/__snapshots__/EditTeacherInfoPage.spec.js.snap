// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditTeacherInfoPage 快照测试 1`] = `
<section class="el-container">
  <main class="el-main">
    <div class="el-page-header">
      <div class="el-page-header__left"><i class="el-icon-back"></i>
        <div class="el-page-header__title">返回</div>
      </div>
      <div class="el-page-header__content">讲师信息</div>
    </div>
    <div class="el-divider el-divider--horizontal">
      <!---->
    </div>
    <div class="el-card is-never-shadow">
      <!---->
      <div class="el-card__body">
        <form class="el-form">
          <div class="el-row">
            <div class="el-form-item" align="center">
              <!---->
              <div class="el-form-item__content">
                <div class="avatar-uploader">
                  <div tabindex="0" class="el-upload el-upload--text"><i class="el-icon-plus avatar-uploader-icon">添加头像</i><input type="file" name="file" class="el-upload__input"></div>
                </div>
                <!---->
              </div>
            </div>
            <div class="el-form-item"><label class="el-form-item__label">讲师名：</label>
              <div class="el-form-item__content">
                <div class="el-input">
                  <!----><input type="text" autocomplete="off" class="el-input__inner">
                  <!---->
                  <!---->
                  <!---->
                  <!---->
                </div>
                <!---->
              </div>
            </div>
          </div>
          <div class="el-row">
            <div class="el-form-item"><label class="el-form-item__label">讲师头衔：</label>
              <div class="el-form-item__content">
                <div class="el-input">
                  <!----><input type="text" autocomplete="off" class="el-input__inner">
                  <!---->
                  <!---->
                  <!---->
                  <!---->
                </div>
                <!---->
              </div>
            </div>
          </div>
          <div class="el-row">
            <div class="el-form-item"><label class="el-form-item__label">讲师联系方式：</label>
              <div class="el-form-item__content">
                <div class="el-input">
                  <!----><input type="text" autocomplete="off" class="el-input__inner">
                  <!---->
                  <!---->
                  <!---->
                  <!---->
                </div>
                <!---->
              </div>
            </div>
          </div>
          <div class="el-row">
            <div class="el-form-item"><label class="el-form-item__label">讲师简介：</label>
              <div class="el-form-item__content">
                <div class="el-textarea"><textarea autocomplete="off" class="el-textarea__inner" style="min-height: -8px; height: -32px;"></textarea>
                  <!---->
                </div>
                <!---->
              </div>
            </div>
          </div>
          <div class="el-form-item">
            <!---->
            <div class="el-form-item__content">
              <div class="el-row is-justify-space-around el-row--flex">
                <div class="el-col el-col-3"><button type="button" class="el-button el-button--primary is-round">
                    <!---->
                    <!----><span>保存</span></button></div>
                <div class="el-col el-col-3"><button type="button" class="el-button el-button--default is-round">
                    <!---->
                    <!----><span>清空</span></button></div>
              </div>
              <!---->
            </div>
          </div>
        </form>
      </div>
    </div>
  </main>
</section>
`;
